src/Bicep.Core/Emit/ResourceDependencyVisitor.cs

```diff
@@ -155,6 +161,65 @@ public override void VisitVariableAccessSyntax(VariableAccessSyntax syntax)
+        /// <param name="resourceSymbol">The referenced resource.</param>
+        private bool IsWeakReference(SyntaxBase syntax, ResourceSymbol resourceSymbol)
+            => resourceSymbol.TryGetResourceType()?.IsAzResource() is true && (
+                IsResourceInfoAccessBase(syntax, resourceSymbol) ||
+                IsResourceFunctionCallBase(syntax) ||
+                IsWithinExplicitParentDeclaration(syntax) ||
+                IsWithinScopeDeclaration(syntax));
+
+        private bool IsResourceInfoAccessBase(SyntaxBase syntax, ResourceSymbol resource)
+            => model.Binder.GetParent(syntax) switch
+            {
+                PropertyAccessSyntax propertyAccess 
+                    => IsResourceInfoAccessBase(resource, propertyAccess.PropertyName.IdentifierName),
+                ArrayAccessSyntax arrayAccess => model.GetTypeInfo(arrayAccess.IndexExpression) switch
+                {
+                    // array access can be used to dereference properties, e.g., resourceSymbolicRef['id']
+                    StringLiteralType indexStr => IsResourceInfoAccessBase(resource, indexStr.RawStringValue),
+                    // it can also dereference a member of a resource collection, e.g., resourceSymbolicRef[0].id
+                    var t when resource.IsCollection && TypeValidator.AreTypesAssignable(t, LanguageConstants.Int)
+                        => IsResourceInfoAccessBase(arrayAccess, resource),
+                    _ => false,
+                },
+                _ => false,
+            };
+
+        private static bool IsResourceInfoAccessBase(ResourceSymbol resource, string propertyName)
+            // if specific top-level properties of an ARM resource are accessed, the compiler will migrate syntax from
+            // the resource declaration or emit a `resourceInfo()` function
+            => ResourceInfoProperties.Contains(propertyName);
+
+        private bool IsResourceFunctionCallBase(SyntaxBase syntax) => model.Binder.GetParent(syntax) switch
+        {
+            InstanceFunctionCallSyntax => true,
+            ArrayAccessSyntax arrayAccess when model.GetSymbolInfo(arrayAccess.BaseExpression) is ResourceSymbol r &&
+                r.IsCollection &&
+                TypeValidator.AreTypesAssignable(model.GetTypeInfo(arrayAccess.IndexExpression), LanguageConstants.Int)
+                => IsResourceFunctionCallBase(arrayAccess),
+            _ => false,
+        };
+
+        private bool IsWithinExplicitParentDeclaration(SyntaxBase syntax)
+            => TryGetCurrentDeclarationTopLevelProperty(LanguageConstants.ResourceParentPropertyName) is { } nonNull &&
+                model.Binder.IsDescendant(syntax, nonNull);
+
+        private bool IsWithinScopeDeclaration(SyntaxBase syntax)
+            => TryGetCurrentDeclarationTopLevelProperty(LanguageConstants.ResourceScopePropertyName) is { } nonNull &&
+                model.Binder.IsDescendant(syntax, nonNull);
```

Comment:
>IsDescendant [](http://example.com/codeflow?start=29&length=12)

My memory is hazy on this one. Does the binder IsDescendant method account for loops for parent and scope scenarios?

Reply:
Yes, because it's only looking at syntax, and anything within a loop body is a descendant of the loop.

Reply:
Makes sense.